Code/Framework/AzToolsFramework/AzToolsFramework/API/ComponentEntitySelectionBus.h

```diff
@@ -8,91 +8,99 @@
+        //! @brief Returns an AABB that encompasses the object.
+        //! @return AABB that encompasses the object.
+        //! @note ViewportInfo may be necessary if the all or part of the object
+        //! stays at a constant size regardless of camera position.
+        virtual AZ::Aabb GetEditorSelectionBoundsViewport([[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo)
         {
-            AZ_Assert(!SupportsEditorRayIntersect(),
+            AZ_Assert(
+                !SupportsEditorRayIntersect(),
                 "Component claims to support ray intersection but GetEditorSelectionBoundsViewport "
                 "has not been implemented in the derived class");
 
             return AZ::Aabb::CreateNull();
         }
 
-        /// @brief Returns true if editor selection ray intersects with the handler.
-        /// @return True if the editor selection ray intersects the handler.
-        /// @note ViewportInfo may be necessary if the all or part of the object
-        /// stays at a constant size regardless of camera position.
+        //! @brief Returns true if editor selection ray intersects with the handler.
+        //! @return True if the editor selection ray intersects the handler.
+        //! @note ViewportInfo may be necessary if the all or part of the object
+        //! stays at a constant size regardless of camera position.
         virtual bool EditorSelectionIntersectRayViewport(
-            const AzFramework::ViewportInfo& /*viewportInfo*/,
-            const AZ::Vector3& /*src*/, const AZ::Vector3& /*dir*/, float& /*distance*/)
+            [[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo,
+            [[maybe_unused]] const AZ::Vector3& src,
+            [[maybe_unused]] const AZ::Vector3& dir,
+            [[maybe_unused]] float& distance)
         {
-            AZ_Assert(!SupportsEditorRayIntersect(),
+            AZ_Assert(
+                !SupportsEditorRayIntersect(),
                 "Component claims to support ray intersection but EditorSelectionIntersectRayViewport "
                 "has not been implemented in the derived class");
 
             return false;
         }
 
-        /// @brief Returns true if the component overrides EditorSelectionIntersectRay method,
-        /// otherwise selection will be based only on AABB test.
-        /// @return True if EditorSelectionIntersectRay method is implemented.
-        virtual bool SupportsEditorRayIntersect() { return false; }
+        //! @brief Returns true if the component overrides EditorSelectionIntersectRay method,
+        //! otherwise selection will be based only on AABB test.
+        //! @return True if EditorSelectionIntersectRay method is implemented.
+        virtual bool SupportsEditorRayIntersect()
+        {
+            return false;
+        }
+
+        virtual bool SupportsEditorRayIntersectViewport([[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo)
```

Comment:
A doxygen comment would be nice for `SupportsEditorRayIntersectViewport` to know the difference with `SupportsEditorRayIntersect`, which says it'll return true if `EditorSelectionIntersectRay` is overridden, is it suppose to be the same for this new function or something else?

Reply:
That's a good call, I'll add some additional info 👍

Reply:
Updated the comment 👍